ts/node/sql.ts

```diff
@@ -3429,42 +3445,48 @@ function cleanUpOldOpengroups() {
 
   db.transaction(() => {
     dropFtsAndTriggers(db);
-    v2Convos.forEach(convo => {
-      const convoId = convo.id;
-      const messagesInConvoBefore = getMessagesCountByConversation(convoId);
-
-      if (messagesInConvoBefore >= maxMessagePerOpengroupConvo) {
-        const minute = 1000 * 60;
-        const sixMonths = minute * 60 * 24 * 30 * 6;
-        const messagesTimestampToRemove = Date.now() - sixMonths;
-        const countToRemove = assertGlobalInstance()
-          .prepare(
-            `SELECT count(*) from ${MESSAGES_TABLE} WHERE serverTimestamp <= $serverTimestamp AND conversationId = $conversationId;`
-          )
-          .get({ conversationId: convoId, serverTimestamp: Date.now() - sixMonths })['count(*)'];
-        const start = Date.now();
-
-        assertGlobalInstance()
-          .prepare(
-            `
-      DELETE FROM ${MESSAGES_TABLE} WHERE serverTimestamp <= $serverTimestamp AND conversationId = $conversationId`
-          )
-          .run({ conversationId: convoId, serverTimestamp: messagesTimestampToRemove }); // delete messages older than sixMonths
-        const messagesInConvoAfter = getMessagesCountByConversation(convoId);
-
-        console.info(
-          `Cleaning ${countToRemove} messages older than 6 months in public convo: ${convoId} took ${Date.now() -
-            start}ms. Old message count: ${messagesInConvoBefore}, new message count: ${messagesInConvoAfter}`
-        );
-
-        const unreadCount = getUnreadCountByConversation(convoId);
-        const convoProps = getConversationById(convoId);
-        if (convoProps) {
-          convoProps.unreadCount = unreadCount;
-          updateConversation(convoProps);
+    if (pruneSetting !== 0) {
+      v2Convos.forEach(convo => {
+        const convoId = convo.id;
+        const messagesInConvoBefore = getMessagesCountByConversation(convoId);
+
+        if (messagesInConvoBefore >= maxMessagePerOpengroupConvo) {
+          const minute = 1000 * 60;
+          const userDefinedMonths = minute * 60 * 24 * 30 * pruneSetting;
+          const messagesTimestampToRemove = Date.now() - userDefinedMonths;
+          const countToRemove = assertGlobalInstance()
+            .prepare(
+              `SELECT count(*) from ${MESSAGES_TABLE} WHERE serverTimestamp <= $serverTimestamp AND conversationId = $conversationId;`
+            )
+            .get({ conversationId: convoId, serverTimestamp: Date.now() - userDefinedMonths })[
```

Comment:
Can you not use the `messagesTimestampToRemove` here?

Reply:
right